lib/inferno/dsl/fhir_resource_validation.rb

```diff
@@ -51,11 +51,39 @@ def url(validator_url = nil)
 
         # Set the IGs that the validator will need to load
         # Example: ["hl7.fhir.us.core#4.0.0"]
-        # @param igs [Array<String>]
+        # @param validator_igs [Array<String>]
         def igs(validator_igs = nil)
-          @igs = validator_igs if validator_igs
+          cli_context.igs(validator_igs) if validator_igs
 
-          @igs
+          cli_context.igs
+        end
+
+        # Set the cliContext used as part of each validation request.
+        # Fields may be passed as either a Hash or block.
+        # Note that all fields included here will be sent directly in requests,
+        # there is no check that the fields are correct.
+        #
+        # @example
+        #   fhir_resource_validator do
+        #     url 'http://example.com/validator'
+        #     cli_context do
+        #       noExtensibleBindingMessages true
+        #       txServer nil
+        #     end
+        #   end
+        #
+        # @param definition [Hash] raw fields to set, optional
+        def cli_context(definition = nil, &)
+          if @cli_context
+            if definition
+              @cli_context.definition.merge!(definition)
```

Comment:
I'm trying to implement this and I may be missing something but even with the merge-with-block approach, if someone calls the `igs` method then `cli_context` with a block, the `instance_eval` with the block will override what's in the definition already.
For example:
```ruby
Validator.new do
        url 'http://example.com'
        igs 'hl7.fhir.us.core#1.0.1'
        cli_context do
          igs ['hl7.fhir.us.core#3.1.1']
        end
      end
```

I could make `CliContext.method_missing` do a merge as well, but it's starting to feel like overkill. I'm ok with saying "if you set the same field in 2 different places, the last one will be the actual value used". The thing I want to avoid is saying "you have to define cliContext before igs" or anything like that

Reply:
Ultimately I added the same logic to CliContext.method_missing

Reply:
> I'm ok with saying "if you set the same field in 2 different places, the last one will be the actual value used". The thing I want to avoid is saying "you have to define cliContext before igs" or anything like that

Yeah, that's reasonable.